javalib/src/main/scala/java/util/ArrayList.scala

```diff
@@ -44,50 +71,134 @@ class ArrayList[E] private (private[ArrayList] val inner: js.Array[E])
   }
 
   def size(): Int =
-    inner.length
+    if (targetPureWasm) {
+      _size
+    } else {
+      innerJS.length
+    }
 
   override def clone(): AnyRef =
-    new ArrayList(inner.jsSlice(0))
+    if (targetPureWasm) {
+      new ArrayList(innerWasm, size())
+    } else {
+      new ArrayList(innerJS.jsSlice(0), size())
```

Comment:
```suggestion
      new ArrayList(innerJS.jsSlice(0), 0)
```
To consistently keep `_size == 0` on JS.